application/src/test/java/org/thingsboard/server/service/state/DefaultDeviceStateServiceTest.java

```diff
@@ -125,4 +140,81 @@ private void process(Map<EntityKeyType, Map<String, TsValue>> latest, long defau
         Assert.assertEquals(5000L, deviceStateData.getState().getInactivityTimeout());
     }
 
+    @Test
+    public void givenIncreaseInactivityTimeoutAndThenStateIsActive() throws Exception {
```

Comment:
The best approach is to have many small corner cases instead of all-in-one cases with confused names.

General cases: Last Inactivity = 0. LastInactivity > 0

IncreaseInactivity for active device . state == no change
IncreaseInactivity for inactive device. state -> active
IncreaseInactivity a little bit for inactive device. state == no change

Decrease inactivity cases both for Last Inactivity = 0. and LastInactivity > 0

On the timeline, we have key points that can happen in different combinations. The initial state can be both active or inactive for any reason before the test event happen.

`LastInactivityTime ... LastActivityTime ... now ... (LastActivityTime + InactivityTimeout)`

Before your changes, some tests have to fail. After your changes, everything will turn **green**!